listenbrainz/labs_api/labs/api/spotify/utils.py

```diff
@@ -0,0 +1,105 @@
+import re
+from enum import Enum
+
+import psycopg2
+
+from flask import current_app
+from unidecode import unidecode
+from psycopg2.extras import execute_values
+from psycopg2.sql import SQL, Identifier
+
+
+class LookupType(Enum):
+    ALL = "combined_lookup_all"
+    WITHOUT_ALBUM = "combined_lookup_without_album"
+
+
+def detune(artist_name: str) -> str:
+    """ Remove commonly used join phrases from artist name """
+    phrases = [" ft ", " feat ", " ft. ", " feat. "]
+    for phrase in phrases:
+        artist_name = artist_name.replace(phrase, " ")
+    return artist_name
+
+
+def query_combined_lookup(column: LookupType, lookups: list[tuple]):
+    """ Lookup track ids for the given lookups in the metadata index using the specified lookup type"""
+    query = SQL("""
+          WITH lookups (idx, value) AS (VALUES %s)
+        SELECT DISTINCT ON ({column})
+               idx, array_agg(track_id ORDER BY score DESC) AS spotify_track_ids
+          FROM lookups
+          JOIN mapping.spotify_metadata_index
+            ON {column} = value
+      GROUP BY {column}, idx      
+    """).format(column=Identifier(column.value))
+
+    with psycopg2.connect(current_app.config["SQLALCHEMY_TIMESCALE_URI"]) as conn, conn.cursor() as curs:
+        execute_values(curs, query, lookups, page_size=len(lookups))
+        result = curs.fetchall()
+        return {row[0]: row[1] for row in result}
+
+
+def perform_one_lookup(column, metadata, generate_lookup):
```

Comment:
Is there a perform_many_lookups? Otherwise maybe just call it perform_lookup?

Reply:
Indeed!